### PR TITLE
[DMS-342] Allow reset credentials in configuration manager

### PR DESCRIPTION
**Pull request by simpat-adam**

### CODE REVIEW

File: src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs [hunk 1]
```diff
@@ -197,4 +214,77 @@ ILogger<ApplicationModule> logger
...
+                    }
+                }
+
+                break;
+            case ApplicationApiClientsResult.FailureUnknown failure:
+                logger.LogError("Error fetching ApiClients: {failure}", failure);
+                return Results.Problem(statusCode: 500);
+        }
+
+        switch (response.Count)
```
Comment (simpat-adam) [inline src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs:279]: Our data model supports many `ApiClient` per `Application` but the AdminApi spec doesn't.

Comment (stephenfuqua) [reply src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs:279]: I'm going to talk to Jason and Vinaya about this. I don't understand the use case of 1-to-many on those database tables.

Comment (stephenfuqua) [reply src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs:279]: For now I'm assuming that we do not need to distinguish `Application` and `ApiClient` from each other, like the `EdFi_Admin` database does.

Comment (simpat-adam) [reply src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs]: After discussion, leaving the many-to-one in the database but only using `First()` in `reset-credential`

**Review by CSR2017 — APPROVED** [review]

**Pull request merged**